com.sap.cloud.lm.sl.cf.client/src/main/java/com/sap/cloud/lm/sl/cf/client/ResilientCloudControllerClient.java

```diff
@@ -3,376 +3,399 @@
     }
 
     @Override
     public void updateApplicationEnv(String appName, Map<String, String> env) {
-        executeWithRetry(() -> super.updateApplicationEnv(appName, env));
+        executeWithRetry(() -> cc.updateApplicationEnv(appName, env));
     }
 
     @Override
     public void updateApplicationEnv(String appName, List<String> env) {
-        executeWithRetry(() -> super.updateApplicationEnv(appName, env));
+        executeWithRetry(() -> cc.updateApplicationEnv(appName, env));
     }
 
     @Override
     public void updateApplicationInstances(String appName, int instances) {
-        executeWithRetry(() -> super.updateApplicationInstances(appName, instances));
+        executeWithRetry(() -> cc.updateApplicationInstances(appName, instances));
     }
 
     @Override
     public void updateApplicationMemory(String appName, int memory) {
-        executeWithRetry(() -> super.updateApplicationMemory(appName, memory));
+        executeWithRetry(() -> cc.updateApplicationMemory(appName, memory));
     }
 
     @Override
     public void updateApplicationServices(String appName, List<String> services) {
-        executeWithRetry(() -> super.updateApplicationServices(appName, services), HttpStatus.NOT_FOUND);
+        executeWithRetry(() -> cc.updateApplicationServices(appName, services), HttpStatus.NOT_FOUND);
     }
 
     @Override
     public void updateApplicationStaging(String appName, Staging staging) {
-        executeWithRetry(() -> super.updateApplicationStaging(appName, staging));
+        executeWithRetry(() -> cc.updateApplicationStaging(appName, staging));
     }
 
     @Override
     public void updateApplicationUris(String appName, List<String> uris) {
-        executeWithRetry(() -> super.updateApplicationUris(appName, uris), HttpStatus.NOT_FOUND);
+        executeWithRetry(() -> cc.updateApplicationUris(appName, uris), HttpStatus.NOT_FOUND);
     }
 
     @Override
     public void updateServiceBroker(CloudServiceBroker serviceBroker) {
-        executeWithRetry(() -> super.updateServiceBroker(serviceBroker));
+        executeWithRetry(() -> cc.updateServiceBroker(serviceBroker));
     }
 
     @Override
     public void updateServicePlanVisibilityForBroker(String name, boolean visibility) {
-        executeWithRetry(() -> super.updateServicePlanVisibilityForBroker(name, visibility));
+        executeWithRetry(() -> cc.updateServicePlanVisibilityForBroker(name, visibility));
     }
 
     @Override
     public void uploadApplication(String appName, File file, UploadStatusCallback callback) throws IOException {
         executeWithRetry(() -> {
             try {
-                super.uploadApplication(appName, file, callback);
+                cc.uploadApplication(appName, file, callback);
             } catch (IOException e) {
-                throw new RuntimeException(e);
+                throw new IllegalArgumentException(e.getMessage(), e);

```

Comment:
Why IllegalArgument and not IllegalState?

Reply:
To have?